src/sif/recost.cc

```diff
@@ -94,10 +95,11 @@ void recost_forward(baldr::GraphReader& reader,
     int time_restrictions_TODO = -1;
     // if its not time dependent set to 0 for Allowed method below
     const uint64_t localtime = offset_time.valid ? offset_time.local_time : 0;
-    // this edge is not allowed
+    // we should evaluate restrictions even if 'ignore_access' flag is true
     if (predecessor != baldr::kInvalidLabel &&
-        !costing.Allowed(edge, label, tile, edge_id, localtime, offset_time.timezone_index,
-                         time_restrictions_TODO)) {
+        (!costing.Allowed(edge, label, tile, edge_id, localtime, offset_time.timezone_index,
+                          time_restrictions_TODO) &&
+         !ignore_access)) {
```

Comment:
move the flag up in the if so that allowed doesnt get called at all

```suggestion
        !ignore_access && !costing.Allowed(edge, label, tile, edge_id, localtime, offset_time.timezone_index,
                          time_restrictions_TODO)) {
```

Reply:
we can't do this because we need to evaluate time restrictions.

Reply:
or we can explicitly call `costing.EvaluateRestrictions` in case `ignore_access = true`

Reply:
the restrictions are based on access though as well is there really a point of checking them? i guess im more thinking we should make the boolean called `throw_if_not_allowed` or something more generic like `strict` to just completely turn off checking allowed at all. maybe we can flip the meaning and say something like `bool allow_all = false`?

Reply:
https://github.com/valhalla/valhalla/blob/master/test/astar.cc#L784 - this test fails if we skip `Allowed` call here

Reply:
hm. what if we back to the first approach where we recost only shortcuts ? - in this case we will save all `time_restrictions` that we calculated for regular edges.
and, when recosting shortcuts, we can 1) use flag to turn off checking `Allowed` or 2) don't use this flag but in case recosting fails - we just don't expand shortcut edge, add shortcut to the final path

Reply:
i personally prefer this implementation as its a lot less complex (no splicing). you have a good point about showing the restrictions. id say just let it as it is and maybe make a note that we have to put the flag second so we can get restriction information.